src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinderProvider.cs

```diff
@@ -20,9 +19,7 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (context.Metadata.IsComplexType &&
-                !context.Metadata.IsCollectionType &&
-                HasDefaultConstructor(context.Metadata.ModelType.GetTypeInfo()))
+            if (context.Metadata.IsComplexType && !context.Metadata.IsCollectionType)
```

Comment:
Previously, the provider would not have returned a binder for _any_ value type because none have a parameterless constructor visible using Reflection. That meant value types with loads of settable properties weren't handled here. Given the trade-offs I mentioned above, I lean toward checking `context.Metadata.ModelType.GetTypeInfo().IsValueType` here.

Separately, we can decide whether this binder should in the future be usable with "complex" value types.